modules/apps/oauth2-provider/oauth2-provider-web/src/main/resources/META-INF/resources/js/OAuth2Client.ts

```diff
@@ -165,9 +182,17 @@ class OAuth2Client {
 			);
 
 			if (cachedTokenData !== null && cachedTokenData !== undefined) {
-				resolve(JSON.parse(cachedTokenData));
+				const cachedToken = JSON.parse(
+					cachedTokenData
+				) as IOAuth2ClientTokenResponse & {
+					expires_after_ms: number;
+				};
```

Comment:
Can we move this type to a declared interface above?

Reply:
Sure

Reply:
changed